src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,1246 @@
+                    f"value {new_processing_resolution}."
+                )
+            denoising_steps = new_denoising_steps
+            ensemble_size = new_ensemble_size
+            processing_resolution = new_processing_resolution
+
+        def maybe_override(new_denoising_steps: int, new_ensemble_size: int, new_processing_resolution: int):
+            nonlocal denoising_steps, ensemble_size, processing_resolution
+            if denoising_steps is None:
+                denoising_steps = new_denoising_steps
+            if ensemble_size is None:
+                ensemble_size = new_ensemble_size
+            if processing_resolution is None:
+                processing_resolution = new_processing_resolution
+
+        # presets logic
+        if preset == "paper" and generator is None:
+            raise ValueError('`preset` value `"paper"` requires `generator` to be set to ensure reproducibility.')
+        if isinstance(self.scheduler, DDIMScheduler):
+            scheduler = "DDIMScheduler"
+            if preset == "fast":
+                preset_override(10, 1, self.optimal_processing_resolution)
+            elif preset == "precise":
+                preset_override(10, 10, self.optimal_processing_resolution)
+            elif preset == "paper":
+                preset_override(50, 10, self.optimal_processing_resolution)
+            else:
+                # closest to fast
+                maybe_override(10, 1, self.optimal_processing_resolution)
+            assert denoising_steps is not None and denoising_steps > 0
+            if denoising_steps < 10:
+                logger.warning(
+                    f"Detected `denoising_steps`={denoising_steps} with DDIMScheduler; at least 10 is recommended. "
+                    f"Consider using the LCM checkpoint for few-step inference."
+                )
+        elif isinstance(self.scheduler, LCMScheduler):
+            scheduler = "LCMScheduler"
+            if preset == "fast":
+                preset_override(1, 1, self.optimal_processing_resolution)
+            elif preset == "precise":
+                preset_override(4, 5, self.optimal_processing_resolution)
+            elif preset == "paper":
+                preset_override(4, 10, self.optimal_processing_resolution)
+            else:
+                # closest to fast
+                maybe_override(1, 1, self.optimal_processing_resolution)
+            assert denoising_steps is not None and denoising_steps > 0
+            if not (1 <= denoising_steps <= 4):
+                logger.warning(
+                    f"Detected `denoising_steps`={denoising_steps} with LCMScheduler; "
+                    f"recommended value is between 1 and 4."
+                )
+        else:
+            raise RuntimeError(f"Unsupported scheduler type: {type(self.scheduler)}.")
+        assert ensemble_size > 0
+        assert processing_resolution >= 0  # 0 for native
+
+        # input checks
```

Comment:
Error checking should be handled in `check_inputs()`.

Reply:
Fixed